csharp/Platform.Data.Doublets/Memory/Split/Generic/SplitMemoryLinksBase.cs

```diff
@@ -799,14 +801,14 @@ public virtual TLink Update(IList<TLink> restrictions, IList<TLink> substitution
                     InternalTargetsTreeMethods.Attach(ref GetLinkIndexPartReference(target).RootAsTarget, linkIndex);
                 }
             }
-            return linkIndex;
+            return handler != null ? handler(restriction, substitution) : Constants.Continue;
```

Comment:
You should use here before and after. After can be constructed at place using current link address, source and target.